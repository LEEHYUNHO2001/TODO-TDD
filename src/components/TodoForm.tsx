import React from 'react';

export const TodoForm = () => <div>TodoForm</div>;
