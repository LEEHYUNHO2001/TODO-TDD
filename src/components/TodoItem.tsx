import styled from '@emotion/styled';
import React from 'react';

import { TodoProps } from '@/types/todoProps';

interface LabelProps {
  checked: boolean;
}

export const TodoItem = ({ todo }: TodoProps) => {
  const { text, done } = todo;

  return (
    <li>
      <input type="checkbox" id={text} checked={done} readOnly />
      <Label htmlFor={text} checked={done}>
        {text}
      </Label>
      <button type="button">삭제</button>
    </li>
  );
};

const Label = styled.label<LabelProps>`
  ${({ checked }) => checked && 'text-decoration: line-through;'}
`;
