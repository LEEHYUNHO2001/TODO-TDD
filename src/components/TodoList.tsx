import React from 'react';

import { TodoListProps } from '@/types/todoListProps';

import { TodoItem } from './TodoItem';

export const TodoList = ({ todos, onRemove }: TodoListProps) => (
  <ul>
    {todos.map(todo => (
      <TodoItem todo={todo} key={todo.id} onRemove={onRemove} />
    ))}
  </ul>
);
