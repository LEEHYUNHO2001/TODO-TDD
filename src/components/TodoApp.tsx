import React, { useCallback, useEffect, useState } from 'react';

import { Todo } from '@/types/todoProps';

import { TodoForm } from './TodoForm';
import { TodoList } from './TodoList';

export const TodoApp = () => {
  const [todos, setTodos] = useState<Todo[]>([]);

  useEffect(() => {
    const data = [
      {
        id: 1,
        text: 'TODO-TDD',
        done: true,
      },
      {
        id: 2,
        text: 'Velog 작성하기',
        done: true,
      },
    ];
    setTodos(data);
  }, []);

  const onInsert = useCallback(
    (value: string) => {
      const newData = {
        id: todos.length + 1,
        text: value,
        done: false,
      };
      setTodos([...todos, newData]);
    },
    [todos]
  );

  const onRemove = useCallback((id: number) => {
    console.log(id, 'onRemove');
  }, []);

  return (
    <div>
      <TodoForm data-testid="helloworld" onInsert={onInsert} />
      <TodoList todos={todos} onRemove={onRemove} />
    </div>
  );
};
