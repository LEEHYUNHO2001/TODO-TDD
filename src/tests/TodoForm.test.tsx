import { render, screen } from '@testing-library/react';

import { TodoForm } from '@/components/TodoForm';

describe('<TodoForm />', () => {
  it('has input and a button', () => {
    const { container } = render(<TodoForm />);
    const input = screen.getByPlaceholderText('할 일을 입력하세요');
    const text = screen.getByText('등록하기');
    expect(input).toBeInTheDocument();
    expect(text).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
