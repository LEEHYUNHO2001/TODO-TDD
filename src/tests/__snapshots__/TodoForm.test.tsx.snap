// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> has input and a button 1`] = `
<div>
  <form>
    <input
      placeholder="할 일을 입력하세요"
    />
    <button
      type="submit"
    >
      등록하기
    </button>
  </form>
</div>
`;
